vpr/src/base/netlist_writer.cpp

```diff
@@ -1595,23 +1606,25 @@ class NetlistWriterVisitor : public NetlistVisitor {
             }
         }
 
-        return std::make_shared<BlackBoxInst>(type_name, inst_name, params, attrs, input_port_conns, output_port_conns, timing_arcs, ports_tsu, ports_tcq);
+        return std::make_shared<BlackBoxInst>(type_name, inst_name, params, attrs, input_port_conns, output_port_conns, timing_arcs, ports_tsu, ports_thld, ports_tcq);
     }
 
     std::shared_ptr<Instance> make_blackbox_instance(const t_pb* atom) {
         const auto& top_pb_route = find_top_pb_route(atom);
         const t_pb_graph_node* pb_graph_node = atom->pb_graph_node;
         const t_pb_type* pb_type = pb_graph_node->pb_type;
 
+        auto& timing_ctx = g_vpr_ctx.timing();
         std::string type_name = pb_type->model->name;
         std::string inst_name = join_identifier(type_name, atom->name);
         std::map<std::string, std::string> params;
         std::map<std::string, std::string> attrs;
         std::map<std::string, std::vector<std::string>> input_port_conns;
         std::map<std::string, std::vector<std::string>> output_port_conns;
         std::vector<Arc> timing_arcs;
-        std::map<std::string, double> ports_tsu;
-        std::map<std::string, double> ports_tcq;
+        std::map<std::string, sequential_port_delay_pair> ports_tsu;
+        std::map<std::string, sequential_port_delay_pair> ports_thld;
+        std::map<std::string, sequential_port_delay_pair> ports_tcq;
 
         //Delay matrix[sink_tnode] -> tuple of source_port_name, pin index, delay
```

Comment:
Good to explain what this is a bit more -- i.e. what is the vector for?  I believe this stores a vector for each sink node, with each element of that vector corresponding to one timing edge with sink_node as its destination.

Reply:
Comment added